src/thread.cpp

```diff
@@ -63,11 +63,18 @@ void zmq::thread_t::start (thread_fn *tfn_, void *arg_, const char *name_)
     _arg = arg_;
     if (name_)
         strncpy (_name, name_, sizeof (_name) - 1);
+
+    // set default stack size to 4MB to avoid std::map stack overflow on x64
+    unsigned int stack = 0;
+#if defined _WIN64
+    stack = 0x400000; 
```

Comment:
If this fix is only relevant for VS2010, shouldn't this be restricted to that version? Sticking with the default if possible seems to be desirable.

Reply:
i think you're right. at the moment I have no way to making the change and I don't know if I will ever have that option.